src/cli.ts

```diff
@@ -28,15 +43,53 @@ async function main(argv: minimist.ParsedArgs) {
   const argOut: string = argv._[1] || argv.output
 
   try {
-    const schema: JSONSchema4 = JSON.parse(await readInput(argIn))
-    const ts = await compile(schema, argIn, argv as Partial<Options>)
-    await writeOutput(ts, argOut)
+    const files = await getFilesToProcess(argIn, argOut, argv as Partial<Options>)
+    await Promise.all(files)
   } catch (e) {
     console.error(whiteBright.bgRedBright('error'), e)
     process.exit(1)
   }
 }
 
+function getFilesToProcess(argIn: string, argOut: string, argv: Partial<Options>): Promise<Promise<void>[]> {
+  return new Promise(async (res, rej) => {
+    try {
+      if (isGlob(argIn)) {
+        const files = await glob(join(process.cwd(), argIn))
+
+        if (files.length === 0) {
+          rej('No files match glob pattern')
+        }
+
+        if (argOut && !existsSync(argOut)) {
+          await mkdirp(argOut)
+        }
+
+        res(files.map(file => processFile(file, {dir: argOut}, argv)))
```

Comment:
Could we make this a little more explicit, and kill some of the boilerplate?

1. Rename `getFilesToProcess` to `processFiles`, update its return type to `Promise<void[]>`, and mark it `async`
2. Remove the `Promise.all` on 47
3. Remove `processFiles`'s explicit `new Promise...` wrapper (lines 55, 56, 73-77)
4. Update L68 to:

```ts
  return Promise.all(
    files.map(file => processFile(file, {dir: argOut}, argv))
  )
```

Also:

1. Can we also remove the duplication between line 68 and 71, by setting `files` to `[argIn]` early when the input is not a directory or a glob?
2. What will happen when the input glob resolves to a file tree with multiple sub-folders that need to be `mkdirp`'d? Can we make sure this works, and add a test case for it? eg.

```
/input
  /a
    /x.json
  /b
    /y.json
/output
```

Where `/output` does not have any sub-folders, we should create both `/output/a` and `/output/b`.

Reply:
I like this idea. I will try and add this next